Source/Core/UICommon/UICommon.cpp

```diff
@@ -137,11 +137,57 @@ void SetUserDirectory(const std::string& custom_path)
 		user_path += DIR_SEP;
 #else
 	if (File::Exists(ROOT_DIR DIR_SEP USERDATA_DIR))
+	{
 		user_path = ROOT_DIR DIR_SEP USERDATA_DIR DIR_SEP;
+	}
 	else
-		user_path = std::string(getenv("HOME") ?
-			getenv("HOME") : getenv("PWD") ?
-			getenv("PWD") : "") + DIR_SEP DOLPHIN_DATA_DIR DIR_SEP;
+	{
+		const char* home = getenv("HOME");
+		if (!home)
+			home = getenv("PWD");
+		if (!home)
+			home = ".";
+		std::string home_path = std::string(home) + DIR_SEP;
+
+#if defined(__APPLE__) || defined(ANDROID)
+		user_path = home_path + DOLPHIN_DATA_DIR DIR_SEP;
```

Comment:
On Android devices, this would end up being set to './sdcard/dolphin-emu/' Which would break the configuration on those devices in obvious ways.